### ISSUE TITLE
Retainer block: Like sunblock for rooms, only better

### ISSUE THREAD

**Issue by Zironic** [Arx-Game/arxcode#70]
It seems like a fairly common complaint about large scenes in general and the assembly of peers in particular is retainer spam. Most of the time people don't intend to spam, they just forget.

This could be prevented with a room flag that prevents retainers from entering or being summoned which could be applied to rooms such as the assembly.

An alternative and probably harder to code solution would be a conditional squelch on retainer emits.

**Comment by TehomCD** [comment]
Yeah, that'd be fine. There's already checks for retainers refusing to enter some rooms, or were at one point, though I don't recall specifics.

**Comment by rozzingit** [comment]
Do note that if you block ALL retainers from entering a room like the Assembly of Peers, you're just gonna end up with messenger spam because no one has their discreet messengers anymore. Maybe if it was a check that blocked by retainer type? Could block guards, champions, animals, and small animals, but maybe leave assistants and spies. I think you'd catch the bulk of spam that way.

**Comment by TehomCD** [comment]
Or making their entrance/exit silent/aggregated. Abbreviate it to `{character_name} and 5 followers enter` or something.

**Comment by emonical** [comment]
Hmm. Maybe something like Schmuckatelli and their entourage, with look schmuckatelli's entourage, like a virtual container of NPCs.

**Comment by TehomCD** [comment]
We already override possessives for look, could make "followers" or "entourage" a special case, sure.
